File: ruby/testcode/hardcodedcreds_001.rb
require_relative 'shared'

# vuln-code-snippet start ruby_hardcoded_db_pass
def connect_to_database(req)
  host = req.param('host')
  user = 'admin'
  conn = PG.connect(host: host, user: user, password: 'Sup3rS3cr3t!')  # vuln-code-snippet vuln-line ruby_hardcoded_db_pass
  result = conn.exec('SELECT version()')
  BenchmarkResponse.ok(result.first['version'])
end
# vuln-code-snippet end ruby_hardcoded_db_pass

File: rust/testcode/hardcodedcreds_001.rs
//! CWE-798: Database connection string with embedded password in const.

// vuln-code-snippet start testcodeHardcodedcreds001
const DB_URL: &str = "postgres://admin:s3cretpass@db:5432/app"; // vuln-code-snippet target-line testcodeHardcodedcreds001

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let table = req.param("table");
    let result = format!("Connected to {} for table {}", DB_URL, table);
    super::shared::BenchmarkResponse::ok(&result)
}
// vuln-code-snippet end testcodeHardcodedcreds001
